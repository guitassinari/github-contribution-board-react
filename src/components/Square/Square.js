import PropTypes from 'prop-types'
import styled from 'styled-components'

const Square = styled.div`
  width: 50%;

  &:after {
    content: "";
    display: block;
    background-color: ${props => props.backgroundColor};
    padding-bottom: 100%;
  }
`

Square.propTypes = {
  backgroundColor: PropTypes.string
}

Square.defaultProps = {
  backgroundColor: "#EBEDF0"
}

export default Square