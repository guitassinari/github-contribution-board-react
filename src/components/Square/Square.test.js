import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer'
import Square from './Square';

describe('Square', () => {
  test('Matches snapshot', () => {
    const component = renderer.create(<Square />);
    const tree = component.toJSON()
  
    expect(tree).toMatchSnapshot();
  });

  describe('When receives backgroundColor prop', () => {
    test('Matches snapshot', () => {
      const component = renderer.create(<Square backgroundColor="#000000" />);
      const tree = component.toJSON()
    
      expect(tree).toMatchSnapshot();
    })
  })
})
