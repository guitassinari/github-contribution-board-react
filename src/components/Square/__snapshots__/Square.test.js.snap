// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square Matches snapshot 1`] = `
<div
  className="sc-bdnylx bpjleA"
/>
`;

exports[`Square When receives backgroundColor prop Matches snapshot 1`] = `
<div
  className="sc-bdnylx eCLRiA"
/>
`;
